modules/sdk-coin-dot/src/dot.ts

```diff
@@ -336,9 +339,19 @@ export class Dot extends BaseCoin {
 
   /**
    * Builds a funds recovery transaction without BitGo
-   * @param params
+   * @param {RecoveryOptions} params parameters needed to construct and
+   * (maybe) sign the transaction
+   *
+   * @returns {DotTx} the serialized transactoin hex string and index
+   * of the address being swept
    */
   async recover(params: RecoveryOptions): Promise<DotTx> {
+    if (!params.startIdx) {
+      params.startIdx = 0;
+    }
+    if (!params.numIteration) {
+      params.numIteration = 20;
+    }
```

Comment:
not a fan of mutating inputs.

can do something like:
```
const startIdx = startIdx ?? 0;
const numIteration = numIteration ?? 20;
```

Reply:
lets also do some input validation.
like params.startIdx >= 0 and is an integer if inputted.
and params.numIteration > 0 and is an integer if inputted